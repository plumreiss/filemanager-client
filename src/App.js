import { Header } from "./components/Header";
import { BaseTable } from "./components/BaseTable";
import { Alert, Container, Row, Spinner } from "react-bootstrap";
import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { fetchFileList } from "./redux/fileSlice";
import { useSelector } from "react-redux";

function App() {
  const dispath = useDispatch();
  const state = useSelector((state) => state.files);

  useEffect(() => {
    (() => {
      dispath(fetchFileList());
    })();
  }, [dispath]);

  let errorMessage = "An unexpected error ocurred";
  if (state.error === "Request failed with status code 404") {
    errorMessage = "No files found";
  } else if (state.error === "Request failed with status code 403") {
    errorMessage = "File is not correctly formatted";
  } else {
    errorMessage = "An unexpected error occurred";
  }

  return (
    <>
      <Header />
      <Container className="my-5">
        <Row className="justify-content-center">
          {state.error && <Alert variant="danger">{errorMessage}</Alert>}
          {state.isLoading && <Spinner data-testid="spinner" />}
          {state.files.length > 0 && <BaseTable files={state.files} />}
        </Row>
      </Container>
    </>
  );
}

export default App;
