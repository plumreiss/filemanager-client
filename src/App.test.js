import configureStore from "redux-mock-store";
import { render } from "@testing-library/react";

import { Provider, useDispatch } from "react-redux";

import App from "./App";

const mockStore = configureStore([]);

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
}));

describe("<App / >", () => {
  it("should be able to render files row", () => {
    const mockDispatch = jest.fn();

    useDispatch.mockReturnValue(mockDispatch);
    const initialState = {
      files: {
        error: "",
        isLoading: false,
        files: [
          {
            file: "test2.csv",
            lines: [
              {
                text: "SvmswXgsimrFzYuiiQrsGIyQ",
                number: 315144,
                hex: "b53f24e6812e8a1d74bcc9159cfb8c19",
              },
            ],
          },
        ],
      },
    };
    const store = mockStore(initialState);

    const component = render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    component.getByText("SvmswXgsimrFzYuiiQrsGIyQ");
    component.getByText("315144");
    component.getByText("b53f24e6812e8a1d74bcc9159cfb8c19");

    expect(component.container).toMatchSnapshot();
  });
});
