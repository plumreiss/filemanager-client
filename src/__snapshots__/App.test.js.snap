// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App / > should be able to render files row 1`] = `
<div>
  <header
    class="container-fluid"
  >
    <div
      class="py-1 text-white row"
      style="background: rgb(255, 102, 102);"
    >
      <h1>
        React Test App
      </h1>
    </div>
  </header>
  <div
    class="my-5 container"
  >
    <div
      class="justify-content-center row"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                File Name
              </th>
              <th>
                Text
              </th>
              <th>
                Number
              </th>
              <th>
                Hex
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                test2.csv
              </td>
              <td>
                SvmswXgsimrFzYuiiQrsGIyQ
              </td>
              <td>
                315144
              </td>
              <td>
                b53f24e6812e8a1d74bcc9159cfb8c19
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
